bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeperTestCase.java

```diff
@@ -219,7 +272,10 @@ private void submit(Runnable operation) {
     }
 
     protected void registerMockEntryForRead(long ledgerId, long entryId, BookieSocketAddress bookieSocketAddress,
-        byte[] entryData, long lastAddConfirmed) {
+        byte[] entryData, long lastAddConfirmed) throws BKException{
+        if (failedBookies.contains(bookieSocketAddress)) {
+            throw  BKException.create(NoBookieAvailableException);
```

Comment:
This is an error, this is a mock setup method, it is not business logic, you are not reading/writing at this point. The error manifests on line 396 and it is the correct behaviour.
I this you should simply drop this 'if' block and than remove the 'throws' clause

Reply:
I think the exception in write path is necessary, it's used to trigger ensemble change, otherwise the ensemble will never be changed.

Reply:
Maybe I am messing up with the diff. I will check out your branch and give you more context.